pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -467,6 +468,26 @@ func (c *Cluster) EnableServiceMonitor(activeDaemon string) error {
 	}
 	serviceMonitor.Spec.NamespaceSelector.MatchNames = []string{c.clusterInfo.Namespace}
 	serviceMonitor.Spec.Selector.MatchLabels = c.selectorLabels(activeDaemon)
+
+	// If the cephcluster is managed by another resource,
+	// add the name of that resource as a label on the ceph metrics.
+	if c.spec.Labels != nil {
```

Comment:
Could you actually factor out this new code block into a method such as `applyMonitoringLabels()`, then add a unit test for that new method?

Reply:
Ack, Sure I will do that.

Reply:
@travisn I updated the PR according to your recommendation. Please review. Thank you.